src/main/java/com/minecolonies/coremod/commands/generalcommands/RandomTeleportCommand.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+        if (SPAWN_NO_TP >= LOWER_BOUNDS)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("Please have an admin raise the maxDistanceFromWorldSpawn number in config."));
+            return;
+        }
+
+        if (!canCommandSenderUseCommand(RTP))
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("Not happenin bro!!, ask an OP to TP you."));
+            return;
+        }
+
+        EntityPlayer playerToTeleport = null;
+
+        if (sender instanceof EntityPlayer)
+        {
+            playerToTeleport = (EntityPlayer) sender;
+        }
+
+        //If the arguments aren't empty, the sender probably wants to teleport another player.
+        if (args.length != 0 && isPlayerOpped(sender, "ctp"))
+        {
+            final World world = FMLCommonHandler.instance().getMinecraftServerInstance().getEntityWorld();
+            playerToTeleport =
+                    ServerUtils.getPlayerFromUUID(FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerProfileCache()
+                            .getGameProfileForUsername(args[0]).getId(), world);
+
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("TPin Player: " + playerToTeleport.getName()));
+        }
+
+        if (playerToTeleport == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+
+        teleportPlayer(sender, playerToTeleport);
+        //.fallDistance is used to cancel out fall damage  basically if you have -5 it will reduce fall damage by 2.5 hearts
+        playerToTeleport.fallDistance=FALL_DISTANCE;
+    }
+
+    /**
+     * Get a random coordinate to teleport to.
+     * @return
+     */
+    private static int getRandCoordinate()
+    {
+        final Random rnd = new Random();
+
+        int x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+
+        /* keeping X out of the spawn radius */
+        while (x > -SPAWN_NO_TP && x < SPAWN_NO_TP)
+        {
+            x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+        }
+
+        return x;
+    }
+
+    /**
+     * Method used to teleport the player.
+     * @param sender           the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private static void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)